internal/target/options.go

```diff
@@ -178,3 +181,11 @@ func WithTargetIds(with []string) Option {
 		o.WithTargetIds = with
 	}
 }
+
+// WithListPermissions enables this repo to optimise resource listing
+// database queries.
```

Comment:
While this does help with optimizing the resource listing, I think it primarily is used to restrict the list results based on this set of permissions.

Reply:
Yep, thanks, addressed in [aa56f90](https://github.com/hashicorp/boundary/pull/2319/commits/aa56f905816e7b4d6b1eb9e04ea5ba85fc61cb20)